compiler/rustc_parse/src/parser/stmt.rs

```diff
@@ -73,13 +73,13 @@ impl<'a> Parser<'a> {
         Ok(Some(if self.token.is_keyword(kw::Let) {
             self.parse_local_mk(lo, attrs, capture_semi, force_collect)?
         } else if self.is_kw_followed_by_ident(kw::Mut) {
```

Comment:
Can you use a [`may_recover`](https://doc.rust-lang.org/beta/nightly-rustc/rustc_parse/parser/struct.Parser.html#method.may_recover) check in all of these failure branches?